src/java.base/share/classes/java/lang/classfile/constantpool/ConstantPoolBuilder.java

```diff
@@ -150,9 +152,10 @@ default Utf8Entry utf8Entry(MethodTypeDesc desc) {
      * returned.
      *
      * @param classDesc the symbolic descriptor for the class
+     * @throws IllegalArgumentException if {@code classDesc} represents a primitive type
      */
     default ClassEntry classEntry(ClassDesc classDesc) {
-        if (classDesc.isPrimitive()) {
+        if (requireNonNull(classDesc).isPrimitive()) {
```

Comment:
It is not clear to me as to whether all the methods in the API deal with null correctly. What we did for FFM, assuming that the classfile API is also null-hostile by default, was to add a single test which would try to call all API methods and looking for NPEs:

https://github.com/openjdk/jdk/blob/2637e8ddc4ffe102418139f501fc0be8e9c5317b/test/jdk/java/foreign/TestNulls.java#L80

This test has saved us many times when adding/changing API methods.

Reply:
Good point, thank.
I've create new RFE https://bugs.openjdk.org/browse/JDK-8317356 because the test would not be so simple to cover all ClassFile API in different situations (for example different builders implementations based on the actual context).